### PR TITLE
Fix gff and scidx parsing

### PR DESCRIPTION
**Pull request by bernt-matthias**
- gff: if there is a `##gff-version` line then respect it
  - it now is analogous to the Gff3 sniffer
  - follow up to https://github.com/galaxyproject/galaxy/pull/12676
- scidx: do not skip the 2nd line

this is fall out of https://github.com/galaxyproject/galaxy/pull/12073 .. found while fixing IUC tools

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

**Comment by gregvonkuster** [comment]
> For ScIdx:
> 
> * should we add a test file to `lib/galaxy/datatypes/test/` ?
> * I think we could greatly simplify the sniffer to just do this: skip the first line iff it starts with `#`, and then check if the next line is equal to `chrom	index	forward	reverse	value`
> 
> Ping @gregvonkuster

I think that's fine.  FWIW, there is a very limited number of tools that use this datatype, and I'm not sure if they are actively used.

File: lib/galaxy/datatypes/interval.py [hunk 1]
```diff
@@ -1466,6 +1466,16 @@ class ScIdx(Tabular):
     count (type int), Reverse strand peak count (type int) and value (type int).
     The value of the 5th 'value' column is the sum of the forward and reverse
     peak count values.
+
+    >>> from galaxy.datatypes.sniff import get_test_fname
+    >>> fname = get_test_fname( 'cntrl_hg19.scidx' )
+    >>> ScIdx().sniff( fname )
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/interval.py]:
```suggestion
    >>> fname = get_test_fname('cntrl_hg19.scidx')
    >>> ScIdx().sniff(fname)
```

File: lib/galaxy/datatypes/interval.py [hunk 2]
```diff
@@ -1466,6 +1466,16 @@ class ScIdx(Tabular):
...
+    True
+    >>> Bed().sniff( fname )
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/interval.py]:
```suggestion
    >>> Bed().sniff(fname)
```

File: lib/galaxy/datatypes/interval.py [hunk 3]
```diff
@@ -1466,6 +1466,16 @@ class ScIdx(Tabular):
...
+    False
+    >>> fname = get_test_fname( 'empty.txt' )
+    >>> ScIdx().sniff( fname )
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/interval.py]:
```suggestion
    >>> fname = get_test_fname('empty.txt')
    >>> ScIdx().sniff(fname)
```

File: lib/galaxy/datatypes/interval.py [hunk 4]
```diff
@@ -1486,49 +1496,31 @@ def sniff_prefix(self, file_prefix):
...
-                if int(forward) + int(reverse) != int(value):
+                if int(line[2]) + int(line[3]) != int(line[4]):
                     return False
-            if count == 100:
-                return True
-            count += 1
-        if count < 100 and count > 0:
+            # just check one data line
+            if count > 2:
+                break
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/interval.py]:
```suggestion
                # just check one data line
                break
```

File: lib/galaxy/datatypes/interval.py [hunk 5]
```diff
@@ -1486,49 +1496,31 @@ def sniff_prefix(self, file_prefix):
...
+        # at least the comment and header are required
+        if count >= 2:
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/interval.py]:
```suggestion
        if count >= 1:
```

Comment (bernt-matthias) [reply lib/galaxy/datatypes/interval.py]: Not sure about this one. This would allow files consisting of only a commen line, or?

Comment (nsoranzo) [reply lib/galaxy/datatypes/interval.py]: Comment (line 0) and header (line 1). I guess that's valid?

File: lib/galaxy/datatypes/interval.py [hunk 6]
```diff
@@ -1486,49 +1496,30 @@ def sniff_prefix(self, file_prefix):
...
-                # EOF
-                if count > 1:
-                    # The second line is always the labels:
-                    # chrom index forward reverse value
-                    # We need at least the column labels and a data line.
-                    return True
-                return False
-            line = line.strip()
+        for count, line in enumerate(file_prefix.line_iterator()):
+            line = line.strip().split()
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/interval.py]:
```suggestion
        for count, line in enumerate(iter_headers(file_prefix, '\t')):
```

**Pull request merged**